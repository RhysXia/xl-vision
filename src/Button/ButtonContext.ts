import React from 'react'
import { ButtonTheme, ButtonVariant } from './Button'

export type ButtonContextProps = {
  theme?: ButtonTheme
  variant?: ButtonVariant
  disableElevation?: boolean
  disableRipple?: boolean
  groupClsPrefix?: string
  size?: string
}

export default React.createContext<ButtonContextProps>({})
