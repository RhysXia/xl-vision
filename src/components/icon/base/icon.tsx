import classnames from 'classnames'
import * as React from 'react'
import { clsPrefix } from '../../_commons/config'

export interface IconProps {
    className?: string
    style?: React.CSSProperties
    children: React.ReactSVGElement
    onClick?: React.MouseEventHandler<HTMLElement>
}

const iconClsPrefix = `${clsPrefix}-icon`

const Icon: React.FunctionComponent<IconProps> = props => {
    const { className, ...others } = props
    const classes = classnames([
        iconClsPrefix
    ], className)
    return (
        <i className={classes} {...others}/>
    )
}

export default Icon
