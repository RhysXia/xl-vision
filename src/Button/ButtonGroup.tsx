import classnames from 'classnames'
import PropTypes from 'prop-types'
import React from 'react'
import { ButtonProps } from './Button'
import ConfigContext from '../ConfigProvider/ConfigContext'
import ButtonContext, { ButtonContextProps } from './ButtonContext'

export interface ButtonGroupProps
  extends Omit<ButtonContextProps, 'groupClsPrefix'>,
    React.HTMLAttributes<HTMLDivElement> {
  children: React.ReactElement<ButtonProps> | Array<React.ReactElement<ButtonProps>>
  clsPrefix?: string
  round?: boolean
}

const ButtonGroup: React.FunctionComponent<ButtonGroupProps> = (props) => {
  const { clsPrefix: rootClsPrefix } = React.useContext(ConfigContext)

  const {
    round,
    className,
    clsPrefix = `${rootClsPrefix}-button-group`,
    disableRipple,
    disableElevation,
    theme,
    variant,
    size,
    ...others
  } = props
  const classes = classnames(
    {
      [clsPrefix]: true,
      [`${clsPrefix}--round`]: round
    },
    className
  )

  return (
    <ButtonContext.Provider
      value={{
        groupClsPrefix: `${clsPrefix}`,
        size,
        disableRipple,
        disableElevation,
        theme,
        variant
      }}
    >
      <div {...others} className={classes} />
    </ButtonContext.Provider>
  )
}

ButtonGroup.displayName = 'ButtonGroup'

ButtonGroup.propTypes = {
  theme: PropTypes.oneOf([
    'default',
    'primary',
    'error',
    'warning',
    'secondary',
    'success',
    'info'
  ]),
  variant: PropTypes.oneOf(['contained', 'text', 'outlined']),
  size: PropTypes.oneOf(['small', 'medium', 'large']),
  disableRipple: PropTypes.bool,
  disableElevation: PropTypes.bool,
  className: PropTypes.string,
  children: PropTypes.oneOfType([
    PropTypes.element.isRequired,
    PropTypes.arrayOf(PropTypes.element.isRequired)
  ]).isRequired,
  clsPrefix: PropTypes.string,
  round: PropTypes.bool
}

export default ButtonGroup
