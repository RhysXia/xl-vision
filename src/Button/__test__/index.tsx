import React from 'react'
import { mount } from 'enzyme'
import Button from '..'
import { Close } from '../../icon'

describe('Button', () => {
  it('基本测试', () => {
    const wrapper = mount(
      <div>
        <Button>button</Button>
        <Button theme='error'>button</Button>
        <Button variant='outlined'>button</Button>
        <Button prefixIcon={<Close />}>button</Button>
        <Button suffixIcon={<Close />}>button</Button>
        <Button prefixIcon={<Close />}></Button>
        <Button suffixIcon={<Close />}></Button>
        <Button long={true}>button</Button>
        <Button loading={true}>button</Button>
        <Button loading={true} prefixIcon={<Close />}></Button>
        <Button loading={true} prefixIcon={<Close />}>
          button
        </Button>
      </div>
    )

    expect(wrapper.render()).toMatchSnapshot()
  })
})

describe('ButtonGroup', () => {
  it('基本测试', () => {
    const wrapper = mount(
      <Button.Group theme='primary' variant='contained'>
        <Button>button</Button>
        <Button theme='error'>button</Button>
        <Button variant='outlined'>button</Button>
        <Button prefixIcon={<Close />}>button</Button>
        <Button suffixIcon={<Close />}>button</Button>
        <Button prefixIcon={<Close />}></Button>
        <Button suffixIcon={<Close />}></Button>
        <Button loading={true}>button</Button>
        <Button loading={true} prefixIcon={<Close />}></Button>
        <Button loading={true} prefixIcon={<Close />}>
          button
        </Button>
      </Button.Group>
    )

    expect(wrapper.render()).toMatchSnapshot()
  })
})
