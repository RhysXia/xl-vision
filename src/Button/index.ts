import Button from './Button'
import ButtonGroup from './ButtonGroup'

export { ButtonProps, ButtonTheme, ButtonVariant } from './Button'
export { ButtonGroupProps } from './ButtonGroup'

const ButtonWithGroup = Button as typeof Button & {
  Group: typeof ButtonGroup
}

ButtonWithGroup.Group = ButtonGroup

export default ButtonWithGroup
