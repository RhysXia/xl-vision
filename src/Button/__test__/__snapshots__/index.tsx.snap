// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button 基本测试 1`] = `
<div>
  <button
    class="xl-base-button xl-button xl-button--variant-contained xl-button--theme-default xl-button--size-medium xl-button--elevation"
    tabindex="0"
  >
    <span
      class="xl-base-button__inner"
    >
      button
    </span>
    <span
      class="xl-ripple"
    />
  </button>
  <button
    class="xl-base-button xl-button xl-button--variant-contained xl-button--theme-error xl-button--size-medium xl-button--elevation"
    tabindex="0"
  >
    <span
      class="xl-base-button__inner"
    >
      button
    </span>
    <span
      class="xl-ripple"
    />
  </button>
  <button
    class="xl-base-button xl-button xl-button--variant-outlined xl-button--theme-default xl-button--size-medium"
    tabindex="0"
  >
    <span
      class="xl-base-button__inner"
    >
      button
    </span>
    <span
      class="xl-ripple"
    />
  </button>
  <button
    class="xl-base-button xl-button xl-button--variant-contained xl-button--theme-default xl-button--size-medium xl-button--elevation"
    tabindex="0"
  >
    <span
      class="xl-base-button__inner"
    >
      <span
        class="xl-button__prefix"
      >
        <i
          class="xl-icon"
        >
          <svg
            style="height: 1em; width: 1em; fill: currentColor; stroke: currentColor;"
            viewBox="0 0 512 512"
          >
            <path
              d="M289.94 256l95-95A24 24 0 00351 127l-95 95-95-95a24 24 0 00-34 34l95 95-95 95a24 24 0 1034 34l95-95 95 95a24 24 0 0034-34z"
            />
          </svg>
        </i>
      </span>
      button
    </span>
    <span
      class="xl-ripple"
    />
  </button>
  <button
    class="xl-base-button xl-button xl-button--variant-contained xl-button--theme-default xl-button--size-medium xl-button--elevation"
    tabindex="0"
  >
    <span
      class="xl-base-button__inner"
    >
      button
      <span
        class="xl-button__suffix"
      >
        <i
          class="xl-icon"
        >
          <svg
            style="height: 1em; width: 1em; fill: currentColor; stroke: currentColor;"
            viewBox="0 0 512 512"
          >
            <path
              d="M289.94 256l95-95A24 24 0 00351 127l-95 95-95-95a24 24 0 00-34 34l95 95-95 95a24 24 0 1034 34l95-95 95 95a24 24 0 0034-34z"
            />
          </svg>
        </i>
      </span>
    </span>
    <span
      class="xl-ripple"
    />
  </button>
  <button
    class="xl-base-button xl-button xl-button--variant-contained xl-button--theme-default xl-button--size-medium xl-button--elevation xl-button--only-icon"
    tabindex="0"
  >
    <span
      class="xl-base-button__inner"
    >
      <i
        class="xl-icon"
      >
        <svg
          style="height: 1em; width: 1em; fill: currentColor; stroke: currentColor;"
          viewBox="0 0 512 512"
        >
          <path
            d="M289.94 256l95-95A24 24 0 00351 127l-95 95-95-95a24 24 0 00-34 34l95 95-95 95a24 24 0 1034 34l95-95 95 95a24 24 0 0034-34z"
          />
        </svg>
      </i>
    </span>
    <span
      class="xl-ripple"
    />
  </button>
  <button
    class="xl-base-button xl-button xl-button--variant-contained xl-button--theme-default xl-button--size-medium xl-button--elevation xl-button--only-icon"
    tabindex="0"
  >
    <span
      class="xl-base-button__inner"
    >
      <i
        class="xl-icon"
      >
        <svg
          style="height: 1em; width: 1em; fill: currentColor; stroke: currentColor;"
          viewBox="0 0 512 512"
        >
          <path
            d="M289.94 256l95-95A24 24 0 00351 127l-95 95-95-95a24 24 0 00-34 34l95 95-95 95a24 24 0 1034 34l95-95 95 95a24 24 0 0034-34z"
          />
        </svg>
      </i>
    </span>
    <span
      class="xl-ripple"
    />
  </button>
  <button
    class="xl-base-button xl-button xl-button--variant-contained xl-button--theme-default xl-button--size-medium xl-button--elevation xl-button--long"
    tabindex="0"
  >
    <span
      class="xl-base-button__inner"
    >
      button
    </span>
    <span
      class="xl-ripple"
    />
  </button>
  <button
    class="xl-base-button xl-base-button--loading xl-button xl-button--variant-contained xl-button--theme-default xl-button--size-medium xl-button--elevation xl-button--loading"
    tabindex="0"
  >
    <span
      class="xl-base-button__inner"
    >
      <span
        class="xl-button__prefix"
      >
        <i
          class="xl-icon xl-button__icon--loading"
        >
          <svg
            style="height: 1em; width: 1em; fill: currentColor; stroke: currentColor;"
            viewBox="0 0 512 512"
          >
            <path
              d="M400 148l-21.12-24.57A191.43 191.43 0 00240 64C134 64 48 150 48 256s86 192 192 192a192.09 192.09 0 00181.07-128"
              fill="none"
              stroke-linecap="round"
              stroke-miterlimit="10"
              stroke-width="32"
            />
            <path
              d="M464 97.42V208a16 16 0 01-16 16H337.42c-14.26 0-21.4-17.23-11.32-27.31L436.69 86.1C446.77 76 464 83.16 464 97.42z"
            />
          </svg>
        </i>
      </span>
      button
    </span>
    <span
      class="xl-ripple"
    />
  </button>
  <button
    class="xl-base-button xl-base-button--loading xl-button xl-button--variant-contained xl-button--theme-default xl-button--size-medium xl-button--elevation xl-button--only-icon xl-button--loading"
    tabindex="0"
  >
    <span
      class="xl-base-button__inner"
    >
      <i
        class="xl-icon xl-button__icon--loading"
      >
        <svg
          style="height: 1em; width: 1em; fill: currentColor; stroke: currentColor;"
          viewBox="0 0 512 512"
        >
          <path
            d="M400 148l-21.12-24.57A191.43 191.43 0 00240 64C134 64 48 150 48 256s86 192 192 192a192.09 192.09 0 00181.07-128"
            fill="none"
            stroke-linecap="round"
            stroke-miterlimit="10"
            stroke-width="32"
          />
          <path
            d="M464 97.42V208a16 16 0 01-16 16H337.42c-14.26 0-21.4-17.23-11.32-27.31L436.69 86.1C446.77 76 464 83.16 464 97.42z"
          />
        </svg>
      </i>
    </span>
    <span
      class="xl-ripple"
    />
  </button>
  <button
    class="xl-base-button xl-base-button--loading xl-button xl-button--variant-contained xl-button--theme-default xl-button--size-medium xl-button--elevation xl-button--loading"
    tabindex="0"
  >
    <span
      class="xl-base-button__inner"
    >
      <span
        class="xl-button__prefix"
      >
        <i
          class="xl-icon xl-button__icon--loading"
        >
          <svg
            style="height: 1em; width: 1em; fill: currentColor; stroke: currentColor;"
            viewBox="0 0 512 512"
          >
            <path
              d="M400 148l-21.12-24.57A191.43 191.43 0 00240 64C134 64 48 150 48 256s86 192 192 192a192.09 192.09 0 00181.07-128"
              fill="none"
              stroke-linecap="round"
              stroke-miterlimit="10"
              stroke-width="32"
            />
            <path
              d="M464 97.42V208a16 16 0 01-16 16H337.42c-14.26 0-21.4-17.23-11.32-27.31L436.69 86.1C446.77 76 464 83.16 464 97.42z"
            />
          </svg>
        </i>
      </span>
      button
    </span>
    <span
      class="xl-ripple"
    />
  </button>
</div>
`;

exports[`ButtonGroup 基本测试 1`] = `
<div
  class="xl-button-group"
>
  <button
    class="xl-base-button xl-button xl-button--variant-contained xl-button--theme-primary xl-button--size-medium xl-button--elevation"
    tabindex="0"
  >
    <span
      class="xl-base-button__inner"
    >
      button
    </span>
    <span
      class="xl-ripple"
    />
  </button>
  <button
    class="xl-base-button xl-button xl-button--variant-contained xl-button--theme-error xl-button--size-medium xl-button--elevation"
    tabindex="0"
  >
    <span
      class="xl-base-button__inner"
    >
      button
    </span>
    <span
      class="xl-ripple"
    />
  </button>
  <button
    class="xl-base-button xl-button xl-button--variant-outlined xl-button--theme-primary xl-button--size-medium"
    tabindex="0"
  >
    <span
      class="xl-base-button__inner"
    >
      button
    </span>
    <span
      class="xl-ripple"
    />
  </button>
  <button
    class="xl-base-button xl-button xl-button--variant-contained xl-button--theme-primary xl-button--size-medium xl-button--elevation"
    tabindex="0"
  >
    <span
      class="xl-base-button__inner"
    >
      <span
        class="xl-button__prefix"
      >
        <i
          class="xl-icon"
        >
          <svg
            style="height: 1em; width: 1em; fill: currentColor; stroke: currentColor;"
            viewBox="0 0 512 512"
          >
            <path
              d="M289.94 256l95-95A24 24 0 00351 127l-95 95-95-95a24 24 0 00-34 34l95 95-95 95a24 24 0 1034 34l95-95 95 95a24 24 0 0034-34z"
            />
          </svg>
        </i>
      </span>
      button
    </span>
    <span
      class="xl-ripple"
    />
  </button>
  <button
    class="xl-base-button xl-button xl-button--variant-contained xl-button--theme-primary xl-button--size-medium xl-button--elevation"
    tabindex="0"
  >
    <span
      class="xl-base-button__inner"
    >
      button
      <span
        class="xl-button__suffix"
      >
        <i
          class="xl-icon"
        >
          <svg
            style="height: 1em; width: 1em; fill: currentColor; stroke: currentColor;"
            viewBox="0 0 512 512"
          >
            <path
              d="M289.94 256l95-95A24 24 0 00351 127l-95 95-95-95a24 24 0 00-34 34l95 95-95 95a24 24 0 1034 34l95-95 95 95a24 24 0 0034-34z"
            />
          </svg>
        </i>
      </span>
    </span>
    <span
      class="xl-ripple"
    />
  </button>
  <button
    class="xl-base-button xl-button xl-button--variant-contained xl-button--theme-primary xl-button--size-medium xl-button--elevation xl-button--only-icon"
    tabindex="0"
  >
    <span
      class="xl-base-button__inner"
    >
      <i
        class="xl-icon"
      >
        <svg
          style="height: 1em; width: 1em; fill: currentColor; stroke: currentColor;"
          viewBox="0 0 512 512"
        >
          <path
            d="M289.94 256l95-95A24 24 0 00351 127l-95 95-95-95a24 24 0 00-34 34l95 95-95 95a24 24 0 1034 34l95-95 95 95a24 24 0 0034-34z"
          />
        </svg>
      </i>
    </span>
    <span
      class="xl-ripple"
    />
  </button>
  <button
    class="xl-base-button xl-button xl-button--variant-contained xl-button--theme-primary xl-button--size-medium xl-button--elevation xl-button--only-icon"
    tabindex="0"
  >
    <span
      class="xl-base-button__inner"
    >
      <i
        class="xl-icon"
      >
        <svg
          style="height: 1em; width: 1em; fill: currentColor; stroke: currentColor;"
          viewBox="0 0 512 512"
        >
          <path
            d="M289.94 256l95-95A24 24 0 00351 127l-95 95-95-95a24 24 0 00-34 34l95 95-95 95a24 24 0 1034 34l95-95 95 95a24 24 0 0034-34z"
          />
        </svg>
      </i>
    </span>
    <span
      class="xl-ripple"
    />
  </button>
  <button
    class="xl-base-button xl-base-button--loading xl-button xl-button--variant-contained xl-button--theme-primary xl-button--size-medium xl-button--elevation xl-button--loading"
    tabindex="0"
  >
    <span
      class="xl-base-button__inner"
    >
      <span
        class="xl-button__prefix"
      >
        <i
          class="xl-icon xl-button__icon--loading"
        >
          <svg
            style="height: 1em; width: 1em; fill: currentColor; stroke: currentColor;"
            viewBox="0 0 512 512"
          >
            <path
              d="M400 148l-21.12-24.57A191.43 191.43 0 00240 64C134 64 48 150 48 256s86 192 192 192a192.09 192.09 0 00181.07-128"
              fill="none"
              stroke-linecap="round"
              stroke-miterlimit="10"
              stroke-width="32"
            />
            <path
              d="M464 97.42V208a16 16 0 01-16 16H337.42c-14.26 0-21.4-17.23-11.32-27.31L436.69 86.1C446.77 76 464 83.16 464 97.42z"
            />
          </svg>
        </i>
      </span>
      button
    </span>
    <span
      class="xl-ripple"
    />
  </button>
  <button
    class="xl-base-button xl-base-button--loading xl-button xl-button--variant-contained xl-button--theme-primary xl-button--size-medium xl-button--elevation xl-button--only-icon xl-button--loading"
    tabindex="0"
  >
    <span
      class="xl-base-button__inner"
    >
      <i
        class="xl-icon xl-button__icon--loading"
      >
        <svg
          style="height: 1em; width: 1em; fill: currentColor; stroke: currentColor;"
          viewBox="0 0 512 512"
        >
          <path
            d="M400 148l-21.12-24.57A191.43 191.43 0 00240 64C134 64 48 150 48 256s86 192 192 192a192.09 192.09 0 00181.07-128"
            fill="none"
            stroke-linecap="round"
            stroke-miterlimit="10"
            stroke-width="32"
          />
          <path
            d="M464 97.42V208a16 16 0 01-16 16H337.42c-14.26 0-21.4-17.23-11.32-27.31L436.69 86.1C446.77 76 464 83.16 464 97.42z"
          />
        </svg>
      </i>
    </span>
    <span
      class="xl-ripple"
    />
  </button>
  <button
    class="xl-base-button xl-base-button--loading xl-button xl-button--variant-contained xl-button--theme-primary xl-button--size-medium xl-button--elevation xl-button--loading"
    tabindex="0"
  >
    <span
      class="xl-base-button__inner"
    >
      <span
        class="xl-button__prefix"
      >
        <i
          class="xl-icon xl-button__icon--loading"
        >
          <svg
            style="height: 1em; width: 1em; fill: currentColor; stroke: currentColor;"
            viewBox="0 0 512 512"
          >
            <path
              d="M400 148l-21.12-24.57A191.43 191.43 0 00240 64C134 64 48 150 48 256s86 192 192 192a192.09 192.09 0 00181.07-128"
              fill="none"
              stroke-linecap="round"
              stroke-miterlimit="10"
              stroke-width="32"
            />
            <path
              d="M464 97.42V208a16 16 0 01-16 16H337.42c-14.26 0-21.4-17.23-11.32-27.31L436.69 86.1C446.77 76 464 83.16 464 97.42z"
            />
          </svg>
        </i>
      </span>
      button
    </span>
    <span
      class="xl-ripple"
    />
  </button>
</div>
`;

exports[`ButtonGroup 基本测试 2`] = `
<div
  class="xl-button-group"
>
  <button
    class="xl-base-button xl-button xl-button--variant-contained xl-button--theme-primary xl-button--size-large xl-button--elevation"
    tabindex="0"
  >
    <span
      class="xl-base-button__inner"
    >
      button
    </span>
    <span
      class="xl-ripple"
    />
  </button>
  <button
    class="xl-base-button xl-button xl-button--variant-contained xl-button--theme-error xl-button--size-large xl-button--elevation"
    tabindex="0"
  >
    <span
      class="xl-base-button__inner"
    >
      button
    </span>
    <span
      class="xl-ripple"
    />
  </button>
  <button
    class="xl-base-button xl-button xl-button--variant-outlined xl-button--theme-primary xl-button--size-large"
    tabindex="0"
  >
    <span
      class="xl-base-button__inner"
    >
      button
    </span>
    <span
      class="xl-ripple"
    />
  </button>
  <button
    class="xl-base-button xl-button xl-button--variant-contained xl-button--theme-primary xl-button--size-large xl-button--elevation"
    tabindex="0"
  >
    <span
      class="xl-base-button__inner"
    >
      <span
        class="xl-button__prefix"
      >
        <i
          class="xl-icon"
        >
          <svg
            style="height: 1em; width: 1em; fill: currentColor; stroke: currentColor;"
            viewBox="0 0 512 512"
          >
            <path
              d="M289.94 256l95-95A24 24 0 00351 127l-95 95-95-95a24 24 0 00-34 34l95 95-95 95a24 24 0 1034 34l95-95 95 95a24 24 0 0034-34z"
            />
          </svg>
        </i>
      </span>
      button
    </span>
    <span
      class="xl-ripple"
    />
  </button>
  <button
    class="xl-base-button xl-button xl-button--variant-contained xl-button--theme-primary xl-button--size-large xl-button--elevation"
    tabindex="0"
  >
    <span
      class="xl-base-button__inner"
    >
      button
      <span
        class="xl-button__suffix"
      >
        <i
          class="xl-icon"
        >
          <svg
            style="height: 1em; width: 1em; fill: currentColor; stroke: currentColor;"
            viewBox="0 0 512 512"
          >
            <path
              d="M289.94 256l95-95A24 24 0 00351 127l-95 95-95-95a24 24 0 00-34 34l95 95-95 95a24 24 0 1034 34l95-95 95 95a24 24 0 0034-34z"
            />
          </svg>
        </i>
      </span>
    </span>
    <span
      class="xl-ripple"
    />
  </button>
  <button
    class="xl-base-button xl-button xl-button--variant-contained xl-button--theme-primary xl-button--size-large xl-button--elevation xl-button--only-icon"
    tabindex="0"
  >
    <span
      class="xl-base-button__inner"
    >
      <i
        class="xl-icon"
      >
        <svg
          style="height: 1em; width: 1em; fill: currentColor; stroke: currentColor;"
          viewBox="0 0 512 512"
        >
          <path
            d="M289.94 256l95-95A24 24 0 00351 127l-95 95-95-95a24 24 0 00-34 34l95 95-95 95a24 24 0 1034 34l95-95 95 95a24 24 0 0034-34z"
          />
        </svg>
      </i>
    </span>
    <span
      class="xl-ripple"
    />
  </button>
  <button
    class="xl-base-button xl-button xl-button--variant-contained xl-button--theme-primary xl-button--size-large xl-button--elevation xl-button--only-icon"
    tabindex="0"
  >
    <span
      class="xl-base-button__inner"
    >
      <i
        class="xl-icon"
      >
        <svg
          style="height: 1em; width: 1em; fill: currentColor; stroke: currentColor;"
          viewBox="0 0 512 512"
        >
          <path
            d="M289.94 256l95-95A24 24 0 00351 127l-95 95-95-95a24 24 0 00-34 34l95 95-95 95a24 24 0 1034 34l95-95 95 95a24 24 0 0034-34z"
          />
        </svg>
      </i>
    </span>
    <span
      class="xl-ripple"
    />
  </button>
  <button
    class="xl-base-button xl-base-button--loading xl-button xl-button--variant-contained xl-button--theme-primary xl-button--size-large xl-button--elevation xl-button--loading"
    tabindex="0"
  >
    <span
      class="xl-base-button__inner"
    >
      <span
        class="xl-button__prefix"
      >
        <i
          class="xl-icon xl-button__icon--loading"
        >
          <svg
            style="height: 1em; width: 1em; fill: currentColor; stroke: currentColor;"
            viewBox="0 0 512 512"
          >
            <path
              d="M400 148l-21.12-24.57A191.43 191.43 0 00240 64C134 64 48 150 48 256s86 192 192 192a192.09 192.09 0 00181.07-128"
              fill="none"
              stroke-linecap="round"
              stroke-miterlimit="10"
              stroke-width="32"
            />
            <path
              d="M464 97.42V208a16 16 0 01-16 16H337.42c-14.26 0-21.4-17.23-11.32-27.31L436.69 86.1C446.77 76 464 83.16 464 97.42z"
            />
          </svg>
        </i>
      </span>
      button
    </span>
    <span
      class="xl-ripple"
    />
  </button>
  <button
    class="xl-base-button xl-base-button--loading xl-button xl-button--variant-contained xl-button--theme-primary xl-button--size-large xl-button--elevation xl-button--only-icon xl-button--loading"
    tabindex="0"
  >
    <span
      class="xl-base-button__inner"
    >
      <i
        class="xl-icon xl-button__icon--loading"
      >
        <svg
          style="height: 1em; width: 1em; fill: currentColor; stroke: currentColor;"
          viewBox="0 0 512 512"
        >
          <path
            d="M400 148l-21.12-24.57A191.43 191.43 0 00240 64C134 64 48 150 48 256s86 192 192 192a192.09 192.09 0 00181.07-128"
            fill="none"
            stroke-linecap="round"
            stroke-miterlimit="10"
            stroke-width="32"
          />
          <path
            d="M464 97.42V208a16 16 0 01-16 16H337.42c-14.26 0-21.4-17.23-11.32-27.31L436.69 86.1C446.77 76 464 83.16 464 97.42z"
          />
        </svg>
      </i>
    </span>
    <span
      class="xl-ripple"
    />
  </button>
  <button
    class="xl-base-button xl-base-button--loading xl-button xl-button--variant-contained xl-button--theme-primary xl-button--size-large xl-button--elevation xl-button--loading"
    tabindex="0"
  >
    <span
      class="xl-base-button__inner"
    >
      <span
        class="xl-button__prefix"
      >
        <i
          class="xl-icon xl-button__icon--loading"
        >
          <svg
            style="height: 1em; width: 1em; fill: currentColor; stroke: currentColor;"
            viewBox="0 0 512 512"
          >
            <path
              d="M400 148l-21.12-24.57A191.43 191.43 0 00240 64C134 64 48 150 48 256s86 192 192 192a192.09 192.09 0 00181.07-128"
              fill="none"
              stroke-linecap="round"
              stroke-miterlimit="10"
              stroke-width="32"
            />
            <path
              d="M464 97.42V208a16 16 0 01-16 16H337.42c-14.26 0-21.4-17.23-11.32-27.31L436.69 86.1C446.77 76 464 83.16 464 97.42z"
            />
          </svg>
        </i>
      </span>
      button
    </span>
    <span
      class="xl-ripple"
    />
  </button>
</div>
`;

exports[`ButtonGroup 基本测试 3`] = `
<div
  class="xl-button-group"
>
  <button
    class="xl-base-button xl-button xl-button--variant-contained xl-button--theme-primary xl-button--size-small xl-button--elevation"
    tabindex="0"
  >
    <span
      class="xl-base-button__inner"
    >
      button
    </span>
    <span
      class="xl-ripple"
    />
  </button>
  <button
    class="xl-base-button xl-button xl-button--variant-contained xl-button--theme-error xl-button--size-small xl-button--elevation"
    tabindex="0"
  >
    <span
      class="xl-base-button__inner"
    >
      button
    </span>
    <span
      class="xl-ripple"
    />
  </button>
  <button
    class="xl-base-button xl-button xl-button--variant-outlined xl-button--theme-primary xl-button--size-small"
    tabindex="0"
  >
    <span
      class="xl-base-button__inner"
    >
      button
    </span>
    <span
      class="xl-ripple"
    />
  </button>
  <button
    class="xl-base-button xl-button xl-button--variant-contained xl-button--theme-primary xl-button--size-small xl-button--elevation"
    tabindex="0"
  >
    <span
      class="xl-base-button__inner"
    >
      <span
        class="xl-button__prefix"
      >
        <i
          class="xl-icon"
        >
          <svg
            style="height: 1em; width: 1em; fill: currentColor; stroke: currentColor;"
            viewBox="0 0 512 512"
          >
            <path
              d="M289.94 256l95-95A24 24 0 00351 127l-95 95-95-95a24 24 0 00-34 34l95 95-95 95a24 24 0 1034 34l95-95 95 95a24 24 0 0034-34z"
            />
          </svg>
        </i>
      </span>
      button
    </span>
    <span
      class="xl-ripple"
    />
  </button>
  <button
    class="xl-base-button xl-button xl-button--variant-contained xl-button--theme-primary xl-button--size-small xl-button--elevation"
    tabindex="0"
  >
    <span
      class="xl-base-button__inner"
    >
      button
      <span
        class="xl-button__suffix"
      >
        <i
          class="xl-icon"
        >
          <svg
            style="height: 1em; width: 1em; fill: currentColor; stroke: currentColor;"
            viewBox="0 0 512 512"
          >
            <path
              d="M289.94 256l95-95A24 24 0 00351 127l-95 95-95-95a24 24 0 00-34 34l95 95-95 95a24 24 0 1034 34l95-95 95 95a24 24 0 0034-34z"
            />
          </svg>
        </i>
      </span>
    </span>
    <span
      class="xl-ripple"
    />
  </button>
  <button
    class="xl-base-button xl-button xl-button--variant-contained xl-button--theme-primary xl-button--size-small xl-button--elevation xl-button--only-icon"
    tabindex="0"
  >
    <span
      class="xl-base-button__inner"
    >
      <i
        class="xl-icon"
      >
        <svg
          style="height: 1em; width: 1em; fill: currentColor; stroke: currentColor;"
          viewBox="0 0 512 512"
        >
          <path
            d="M289.94 256l95-95A24 24 0 00351 127l-95 95-95-95a24 24 0 00-34 34l95 95-95 95a24 24 0 1034 34l95-95 95 95a24 24 0 0034-34z"
          />
        </svg>
      </i>
    </span>
    <span
      class="xl-ripple"
    />
  </button>
  <button
    class="xl-base-button xl-button xl-button--variant-contained xl-button--theme-primary xl-button--size-small xl-button--elevation xl-button--only-icon"
    tabindex="0"
  >
    <span
      class="xl-base-button__inner"
    >
      <i
        class="xl-icon"
      >
        <svg
          style="height: 1em; width: 1em; fill: currentColor; stroke: currentColor;"
          viewBox="0 0 512 512"
        >
          <path
            d="M289.94 256l95-95A24 24 0 00351 127l-95 95-95-95a24 24 0 00-34 34l95 95-95 95a24 24 0 1034 34l95-95 95 95a24 24 0 0034-34z"
          />
        </svg>
      </i>
    </span>
    <span
      class="xl-ripple"
    />
  </button>
  <button
    class="xl-base-button xl-base-button--loading xl-button xl-button--variant-contained xl-button--theme-primary xl-button--size-small xl-button--elevation xl-button--loading"
    tabindex="0"
  >
    <span
      class="xl-base-button__inner"
    >
      <span
        class="xl-button__prefix"
      >
        <i
          class="xl-icon xl-button__icon--loading"
        >
          <svg
            style="height: 1em; width: 1em; fill: currentColor; stroke: currentColor;"
            viewBox="0 0 512 512"
          >
            <path
              d="M400 148l-21.12-24.57A191.43 191.43 0 00240 64C134 64 48 150 48 256s86 192 192 192a192.09 192.09 0 00181.07-128"
              fill="none"
              stroke-linecap="round"
              stroke-miterlimit="10"
              stroke-width="32"
            />
            <path
              d="M464 97.42V208a16 16 0 01-16 16H337.42c-14.26 0-21.4-17.23-11.32-27.31L436.69 86.1C446.77 76 464 83.16 464 97.42z"
            />
          </svg>
        </i>
      </span>
      button
    </span>
    <span
      class="xl-ripple"
    />
  </button>
  <button
    class="xl-base-button xl-base-button--loading xl-button xl-button--variant-contained xl-button--theme-primary xl-button--size-small xl-button--elevation xl-button--only-icon xl-button--loading"
    tabindex="0"
  >
    <span
      class="xl-base-button__inner"
    >
      <i
        class="xl-icon xl-button__icon--loading"
      >
        <svg
          style="height: 1em; width: 1em; fill: currentColor; stroke: currentColor;"
          viewBox="0 0 512 512"
        >
          <path
            d="M400 148l-21.12-24.57A191.43 191.43 0 00240 64C134 64 48 150 48 256s86 192 192 192a192.09 192.09 0 00181.07-128"
            fill="none"
            stroke-linecap="round"
            stroke-miterlimit="10"
            stroke-width="32"
          />
          <path
            d="M464 97.42V208a16 16 0 01-16 16H337.42c-14.26 0-21.4-17.23-11.32-27.31L436.69 86.1C446.77 76 464 83.16 464 97.42z"
          />
        </svg>
      </i>
    </span>
    <span
      class="xl-ripple"
    />
  </button>
  <button
    class="xl-base-button xl-base-button--loading xl-button xl-button--variant-contained xl-button--theme-primary xl-button--size-small xl-button--elevation xl-button--loading"
    tabindex="0"
  >
    <span
      class="xl-base-button__inner"
    >
      <span
        class="xl-button__prefix"
      >
        <i
          class="xl-icon xl-button__icon--loading"
        >
          <svg
            style="height: 1em; width: 1em; fill: currentColor; stroke: currentColor;"
            viewBox="0 0 512 512"
          >
            <path
              d="M400 148l-21.12-24.57A191.43 191.43 0 00240 64C134 64 48 150 48 256s86 192 192 192a192.09 192.09 0 00181.07-128"
              fill="none"
              stroke-linecap="round"
              stroke-miterlimit="10"
              stroke-width="32"
            />
            <path
              d="M464 97.42V208a16 16 0 01-16 16H337.42c-14.26 0-21.4-17.23-11.32-27.31L436.69 86.1C446.77 76 464 83.16 464 97.42z"
            />
          </svg>
        </i>
      </span>
      button
    </span>
    <span
      class="xl-ripple"
    />
  </button>
</div>
`;
